pytket_qir/parser.py

```diff
@@ -104,31 +96,31 @@
 }
 
 
+_RUNTIME_FUNC = [FuncName.INT.value, FuncName.BOOL.value, FuncName.RES.value]
+
+
 class QirParser:
     """A parser class to return a pytket circuit from a QIR file."""
 
     def __init__(
         self,
-        file_path: str,
+        qir_module: QirModule,
         gateset: Optional[CustomGateSet] = None,
         wasm_handler: Optional[WasmFileHandler] = None,
         wasm_int_type: types.Int = types.Int(32),
         qir_int_type: types.Int = types.Int(64),
     ) -> None:
-        self.module = QirModule(file_path)
+        self.module: QirModule = qir_module
         self.gateset: CustomGateSet = gateset if gateset is not None else PYQIR_GATES
         self.wasm_handler = wasm_handler
         self.wasm_int_type = wasm_int_type
         self.qir_int_type = qir_int_type
         self.qubits = self.get_required_qubits()
         self.bits = self.get_required_results()
-        self.set_cregs: Dict[str, BitRegister] = {}  # Keep track of set registers.
+        self.ssa_vars: Dict[str, BitRegister] = {}  # Log of set ssa variables.
```

Comment:
Are this the big registers for the conditions or is this something else?

Reply:
Not necessarily. These are maps from SSA variable names to the corresponding `pytket` register. It is not restricted to only conditions but any SSA variable (for instance as the result of some arithmetic operation). In the `QirConverter`, I filter this map to see if there is one with the right SSA variable name for the condition.